README.md

```diff
@@ -32,20 +855,1218 @@ The written tests available in the directory [`test`](./test) achieve a test cov
+  "function computeCreate2Address(bytes32 salt, bytes32 initCodeHash, address deployer) pure returns (address computedAddress)",
+  "function computeCreate3Address(bytes32 salt, address deployer) pure returns (address computedAddress)",
+  "function computeCreate3Address(bytes32 salt) view returns (address computedAddress)",
+  "function computeCreateAddress(uint256 nonce) view returns (address computedAddress)",
+  "function computeCreateAddress(address deployer, uint256 nonce) view returns (address computedAddress)",
+  "function deployCreate(bytes initCode) payable returns (address newContract)",
+  "function deployCreate2(bytes32 salt, bytes initCode) payable returns (address newContract)",
+  "function deployCreate2(bytes initCode) payable returns (address newContract)",
+  "function deployCreate2AndInit(bytes32 salt, bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values, address refundAddress) payable returns (address newContract)",
+  "function deployCreate2AndInit(bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values) payable returns (address newContract)",
+  "function deployCreate2AndInit(bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values, address refundAddress) payable returns (address newContract)",
+  "function deployCreate2AndInit(bytes32 salt, bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values) payable returns (address newContract)",
+  "function deployCreate2Clone(bytes32 salt, address implementation, bytes data) payable returns (address proxy)",
+  "function deployCreate2Clone(address implementation, bytes data) payable returns (address proxy)",
+  "function deployCreate3(bytes initCode) payable returns (address newContract)",
+  "function deployCreate3(bytes32 salt, bytes initCode) payable returns (address newContract)",
+  "function deployCreate3AndInit(bytes32 salt, bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values) payable returns (address newContract)",
+  "function deployCreate3AndInit(bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values) payable returns (address newContract)",
+  "function deployCreate3AndInit(bytes32 salt, bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values, address refundAddress) payable returns (address newContract)",
+  "function deployCreate3AndInit(bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values, address refundAddress) payable returns (address newContract)",
+  "function deployCreateAndInit(bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values) payable returns (address newContract)",
+  "function deployCreateAndInit(bytes initCode, bytes data, (uint256 constructorAmount, uint256 initCallAmount) values, address refundAddress) payable returns (address newContract)",
```

Comment:
Interesting, viem ABIs can be defined with structs, e.g.
```javascript
"struct Values { uint256 constructorAmount; uint256 initCallAmount; }",
"function deployCreateAndInit(bytes initCode, bytes data, Values memory values) payable returns (address newContract)",
"function deployCreateAndInit(bytes initCode, bytes data, Values memory values, address refundAddress) payable returns (address newContract)",
```

so I wonder why it didn't come out this way.  I forget how I generated [this `Multicall3` one](https://github.com/mds1/multicall3-frontend/blob/7df0684d01cafa57cc729329f35c1b6ce2a9d8b5/src/lib/constants.ts#L246-L267) to get it in that format. Not a big issue though

Reply:
I used your script [here](https://[github.com/mds1/evm-diff/blob/main/script/json-abi-to-viem-human-readable.ts](https://github.com/mds1/evm-diff/blob/main/script/json-abi-to-viem-human-readable.ts). In your case, the struct forms an array, maybe because of this you generated the struct? Can you please confirm that I generated the correct viem ABI using your script. Otherwise, paste it as suggested change and I will commit.

Reply:
Interestingly, the viem guys also imply [here](https://github.com/wagmi-dev/viem/issues/1406#issuecomment-1783916066) to use it as I generated it; so I guess it's fine:

```md
That is an Ethers flavored ABI. Viem only supports generic flavored ABIs. To fix this, simply find and replace the word ”tuple” with ””.
```

Reply:
Agreed this is fine 👌